templates/partials/actionbar.html

```diff
@@ -44,19 +44,20 @@
 
       <div class="col-sm-4 general-actions left-pane" ng-show="currentTab === 'contacts'">
         <div class="actions dropup">
-          <a class="mm-icon mm-icon-inverse mm-icon-caption" ui-sref="contacts.add()" ng-show="facilities.length">
+          <a class="mm-icon mm-icon-inverse mm-icon-caption" ng-show="actionBar.left.userFacilityId" ui-sref="contacts.addChild({ type: actionBar.left.userChildPlace.type, parent_id: actionBar.left.userFacilityId })">
             <span class="fa-stack">
-              <i class="fa fa-building fa-stack-1x"></i>
+              <i class="fa {{actionBar.left.userChildPlace.icon}} fa-stack-1x"></i>
               <i class="fa fa-plus fa-stack-1x"></i>
             </span>
-            <p translate>Add place</p>
+            <p translate>action.{{actionBar.left.userChildPlace.type}}.add</p>
           </a>
-          <a class="mm-icon mm-icon-inverse mm-icon-caption" ui-sref="contacts.add()" ng-show="!facilities.length">
+
+          <a class="mm-icon mm-icon-inverse mm-icon-caption" ui-sref="contacts.add()" ng-show="!actionBar.left.userFacilityId">
             <span class="fa-stack">
-              <i class="fa fa-group fa-stack-1x"></i>
+              <i class="fa fa-building fa-stack-1x"></i>
               <i class="fa fa-plus fa-stack-1x"></i>
             </span>
-            <p translate>action.clinic.add</p>
+            <p translate>Add place</p>
```

Comment:
Yes! It's in the case where `!actionBar.left.userFacilityId` : if you don't know the user's facility_id, fall back on a generic "Add Place" button.